internal/provider/provider.go

```diff
@@ -246,6 +246,12 @@ func NullifyPlanFields[M ComponentModel](plan *M, schema resourceSchema.Schema)
 			panic(fmt.Errorf("Unsupported NullifyPlanFields type \"%T\" for field: \"%s\" in schema \"%s\"", schemaFieldType, schemaFieldName, schema.Description))
 		}
 
-		structField.Set(nullValue)
+		// Do not attempt to assign the null value if the field is not of the
+		// same type.
+		if reflect.TypeOf(nullValue) == structField.Type() {
+			structField.Set(nullValue)
+		} else {
+			continue
+		}
```

Comment:
This shouldn't be needed.  If we don't want to use `NullifyPlanFields` for a certain type, then don't tag it with `user_config:true`.  Plus, we don't want this to fail silently--when it errors we know we need to look at something closer.

If anything, this should `panic` with a specific message instead of `continue`.

Reply:
Fixed. Should we call it nullify_check or something to make it apparent that this is what the tag does?